src/main.cpp

```diff
@@ -1750,7 +1750,7 @@ void clear_things_and_persons_data(void)
         thing->mappos.x.val = subtile_coord_center(gameadd.map_subtiles_x/2);
         thing->mappos.y.val = subtile_coord_center(gameadd.map_subtiles_y/2);
     }
-    for (i=0; i < CREATURES_COUNT; i++)
```

Comment:
This one needs to stay original too. When clearing things, you want to clear all things, not just the things as set in the current rule.